src/main/java/bio/terra/service/rawls/RawlsClient.java

```diff
@@ -40,7 +40,7 @@ public WorkspaceResponse getWorkspace(UUID workspaceId, AuthenticatedUserRequest
     try {
       ResponseEntity<WorkspaceResponse> workspaceCall =
           restTemplate.exchange(
-              String.format("%s/api/workspaces/id/%s", rawlsConfiguration.basePath(), workspaceId),
+              getWorkspaceEndpoint(workspaceId),
```

Comment:
I'm curious, is there retry logic that rest template brings along? I'm thinking of retrying on 5xx, etc.

Reply:
It looks like there is via `spring-retry`, but TDR doesn't presently use it.

https://www.vincenzoracca.com/en/blog/framework/spring/resttemplate-retry/

However, rather than investing further into [RestTemplate](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html) it's likely more valuable to migrate over to WebClient, since RestTemplate is now in maintenance mode.  @pshapiro4broad, you've mentioned this in a few PR reviews IIRC.

Reply:
Yes `WebClient` is the new API, although I seem to recall there was some reason why it wasn't all that great either.

When I added code to catalog to call rawls APIs I used an openapi client instead of writing the API calls directly so we didn't need to use `RestTemplate` or `WebClient`. Although there workarounds needed as the actual APIs don't match the openapi spec. https://github.com/DataBiosphere/terra-data-catalog/blob/main/integration/src/main/java/scripts/client/RawlsClient.java#L70